modular_zubbers/code/modules/status_indicators/status_indicator.dm

```diff
@@ -185,22 +156,9 @@ GLOBAL_LIST_INIT(potential_indicators, list(
 	// In /mob/living's `update_transform()`, the sprite is horizontally shifted when scaled up, so that the center of the sprite doesn't move to the right.
 	// Because of that, this adjustment needs to happen with the future indicator row as well, or it will look bad.
 	current_x_position -= 16 * (icon_scale - DEFAULT_MOB_SCALE)
-
-	// Now the indicator row can actually be built.
-	for(var/all_indicators in status_indicators)
-		var/image/indicator = all_indicators
-
-		// This is a semi-HUD element, in a similar manner as medHUDs, in that they're 'above' everything else in the world,
-		// but don't pierce obfuscation layers such as blindness or darkness, unlike actual HUD elements like inventory slots.
-		indicator.plane = RENDER_PLANE_GAME
-		indicator.layer = STATUS_LAYER
-		indicator.appearance_flags = PIXEL_SCALE|TILE_BOUND|NO_CLIENT_COLOR|RESET_COLOR|RESET_ALPHA|RESET_TRANSFORM|KEEP_APART
-		indicator.pixel_y = y_offset
-		indicator.pixel_x = current_x_position
-		my_carbon_mob.add_overlay(indicator)
-		// Adding the margin space every time saves a conditional check on the last iteration,
-		// and it won't cause any issues since no more icons will be added, and the var is not used for anything else.
-		current_x_position += STATUS_INDICATOR_ICON_X_SIZE + STATUS_INDICATOR_ICON_MARGIN
+	this_indicator.appearance_flags |= (KEEP_TOGETHER|RESET_COLOR|RESET_TRANSFORM)
+	my_carbon_mob.vis_contents |= this_indicator
```

Comment:
if we're adding something to a mob's vis_contents we need to be removing it in Destroy() or it's going to hard delete